src/Voron/Data/BTrees/Tree.MultiTree.cs

```diff
@@ -30,110 +31,304 @@ namespace Voron.Data.BTrees
+                }
+            }
+        }
+        
+        public bool IsMultiValueTree { get; set; }
 
+        /// <summary>
+        /// Adds multiple values to the multi-tree.
+        /// </summary>
+        /// <param name="key">The key associated with the values.</param>
+        /// <param name="values">The values to insert. Requirements: must be sorted and deduplicated.</param>
+        public void MultiBulkAdd(Slice key, ReadOnlySpan<Slice> values)
+        {
+            if (values.Length <= 1)
+            {
+                if (values.Length == 1)
+                    MultiAdd(key, values[0]);
+                return;
+            }
+            
             int maxNodeSize = Llt.DataPager.NodeMaxSize;
-            if (value.Size > maxNodeSize)
-                throw new ArgumentException("Cannot add a value to child tree that is over " + maxNodeSize + " bytes in size", nameof(value));
-            if (value.Size == 0)
-                throw new ArgumentException("Cannot add empty value to child tree");
-
+            EnsureValuesAreSorted(values);
+            ValidateValuesForMultiBulkAdd(maxNodeSize, values);
+            
+            // This is a new tree, we've to put a flag in the header.
             if ((State.Header.Flags & TreeFlags.MultiValueTrees) != TreeFlags.MultiValueTrees)
             {
                 ref var state = ref State.Modify();
                 state.Flags |= TreeFlags.MultiValueTrees;
             }
 
             var page = FindPageFor(key, out _);
-            if (page == null || page.LastMatch != 0)
+            if (page is not { LastMatch: 0 }) // Key is not in the tree. Use an optimized path for this scenario.
             {
-                MultiAddOnNewValue(key, value, maxNodeSize);
+                MultiBulkAddOnNewValue(key, values, maxNodeSize);
                 return;
             }
 
-            page = ModifyPage(page);
+            ContextBoundNativeList<int> newItems;
+            {
+                var readonlyKeyItem = page.GetNode(page.LastSearchPosition);
+                if (readonlyKeyItem->Flags == TreeNodeFlags.MultiValuePageRef)
+                {
 
-            var item = page.GetNode(page.LastSearchPosition);
+                    // We have an inner tree for the key. Since our values are sorted,
+                    // we can iterate over the elements and add them one by one.
+                    // As the page is cached, it is already loaded into memory. 
+                    // Therefore, this path could be further optimized by batching them into a single page.
+                    var existingTree = OpenMultiValueTree(key, readonlyKeyItem);
+                    for (int i = 0; i < values.Length; i++)
+                    {
+                        if (existingTree.Exists(values[i]))
+                            continue;
+                        
+                        existingTree.DirectAdd(values[i], 0,out _).Dispose();
```

Comment:
Why are you doing the `Exists` check? That requires two tree searches, no?
Why not just call `DirectAdd` and let it be no op when size is `0` and it already exists, etc.